include/unifex/get_exception_ptr.hpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/tag_invoke.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <memory>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex
+{
+  inline const struct _get_exception_ptr_cpo {
+    template <typename ErrorCode>
+    constexpr auto operator()(ErrorCode&& error) const noexcept
+        -> tag_invoke_result_t<_get_exception_ptr_cpo, ErrorCode> {
+      return unifex::tag_invoke(*this, (ErrorCode &&) error);
+    }
+  } get_exception_ptr{};
```

Comment:
And I would consider naming this `as_exception_ptr`.

Reply:
I also realized that there is an already existing `make_exception_ptr` CPO, that is commented to be a faster may to create an exception ptr than `std::make_exception_ptr`.
Maybe it could also be used for that purpose instead of creating a new CPO ?

Reply:
I don't think we want to change the meaning of `make_exception_ptr` or make it a customization point, but perhaps after you add `as_exception_ptr`, we can fold the behavior of `make_exception_ptr` into it and remove `make_exception_ptr`.